app/models/course/experience_points_disbursement.rb

```diff
@@ -14,21 +14,31 @@ class Course::ExperiencePointsDisbursement
   #   @return [Course]
   attr_accessor :course
 
+  # @!attribute [r] group
+  #   The group that this disbursement is for. nil is returned if no group is specified.
+  #   @return [Course::Group|nil]
+  attr_reader :group
+
   # @!attribute [r] experience_points_records
   #   Experience points records that will potentially be created.
   #   @return [Array<Course::ExperiencePointsRecords>]
   attr_reader :experience_points_records
 
   validates :reason, presence: true
 
-  # Given a course, builds an empty experience points record for each of its approved students.
+  # Builds an empty experience points record for each student in a course group.
+  # If no group_id is given, records are built for all approved students in the course.
   #
+  # @param [Course] current_course The course this disbursement is for
+  # @param [Integer] group_id The group this disbursement is for
   # @return [Array<Course::ExperiencePointsRecords>] The newly built experience points records
-  def build(current_course)
+  def build(current_course, group_id)
     @course = current_course
-    @experience_points_records = course.course_users.students.with_approved_state.map do |student|
-      student.experience_points_records.build
-    end
+    @group = group_id && course.groups.find_by(id: group_id)
+    @experience_points_records = \
+      (group || course).course_users.students.with_approved_state.map do |student|
```

Comment:
You should return normal users in the group, there are group roles. 

And let's standardise and remove `with_approved_state`, group component should make sure all users in it are approved.